### ISSUE TITLE
python module files erroneously have shabangs

### ISSUE THREAD

**Issue by paulwouters** [beartype/beartype#331]
While packaging this up for fedora, it shows me 127 shebang warnings for modules, eg:

python3-beartype.noarch: E: non-executable-script /usr/lib/python3.12/site-packages/beartype/__init__.py 644 /usr/bin/env python3
python3-beartype.noarch: E: non-executable-script /usr/lib/python3.12/site-packages/beartype/_cave/_caveabc.py 644 /usr/bin/env python3
python3-beartype.noarch: E: non-executable-script /usr/lib/python3.12/site-packages/beartype/_cave/_cavefast.py 644 /usr/bin/env python3
python3-beartype.noarch: E: non-executable-script /usr/lib/python3.12/site-packages/beartype/_cave/_cavemap.py 644 /usr/bin/env python3
python3-beartype.noarch: E: non-executable-script /usr/lib/python3.12/site-packages/beartype/_check/_checksnip.py 644 /usr/bin/env python3

Those files are not supposed to have the shebangs. It wouldn't do much anyway since the files are mode 644.

proposed fix: remove the shebang lines

**Comment by leycec** [comment]
**...lolbro.** With all due respect, Fedora *basically* has no idea what it's talking about. Numerous IDEs, linters, documentation generators, and other automated devtools throughout the open-source ecosystem leverage the shebang as a filetype. There's literally no advantage and considerable disadvantage to removing shebangs. Indeed, these popular open-source Python packages behave in the exact same manner as @beartype:

* `css_parser`.
* `docutils` (i.e., Sphinx).
* `dill`.
* `gentoolkit`.
* `html5_parser`.
* `mechanize`.
* `mpmath`.
* `serial`.
* `svgwrite`.

And the list just goes on *and* on. Are Sphinx and `dill` bad? No. Clearly not. They're good and, indeed, necessary. Is @beartype bad? By the same argument: "No. Clearly not. It's possibly good and possibly even necessary." So, we're agreed that shebangs have no correlation whatsoever with code quality. I'd consider their removal a net negative rather than a net positive.

That said, as a Fedora packager, you're subject to Fedora constraints. I acknowledge this unpleasant reality. Personally, I'd rather not throw scarce open-source volunteer hours at something I regard as a purely subjective stylistic choice. But you might not have a choice. If this is the case, a one-liner resembling the following *should* get you where you need to be – in theory, anyway:

```bash
find beartype/ -type f -print0 | xargs -0 sed -i -e '/^\s*#!/d'
```

Theoretically, that unconditionally strips *all* shebangs from `beartype/`. Theoretically, performing that one-liner in your Fedora package script should suffice to silence Fedora linter complaints. Maybe?

## tl;dr

I really like shebangs. @beartype exists purely to amuse and titillate me. Removing shebangs would remove  amusement and titillation. Unless anybody else complains about shebangs, I'd rather preserve them as is.

Let's quietly close this. If somebody else complains about shebangs, I'll reluctantly reopen this and remove them at that time. With any luck, nobody else will complain about shebangs. I'm crossing my paws here. :feet:
